diff --git a/node_modules/react-native/Libraries/Blob/Blob.js b/node_modules/react-native/Libraries/Blob/Blob.js
index 99d5e59..1c35f3b 100644
--- a/node_modules/react-native/Libraries/Blob/Blob.js
+++ b/node_modules/react-native/Libraries/Blob/Blob.js
@@ -105,6 +105,12 @@ class Blob {
       blobId: this.data.blobId,
       offset,
       size,
+      /* Since `blob.slice()` creates a new view onto the same binary
+       * data as the original blob, we should re-use the same collector
+       * object so that the underlying resource gets deallocated when
+       * the last view into the data is released, not the first.
+       */
+      __collector: this.data.__collector,
     });
   }
 